django/http/request.py

```diff
@@ -280,28 +283,35 @@ def _load_post_and_files(self):
             self._mark_post_parse_error()
             return
 
-        if self.content_type == 'multipart/form-data':
-            if hasattr(self, '_body'):
-                # Use already read data
-                data = BytesIO(self._body)
-            else:
-                data = self
-            try:
+        try:
```

Comment:
Before it looks like:
```python
if self.content_type == 'multipart/form-data':
    if hasattr(self, '_body'):
        # Use already read data
        data = BytesIO(self._body)
    else:
        data = self
    try:
        self._post, self._files = self.parse_file_upload(self.META, data)
    except MultiPartParserError:
        # An error occurred while parsing POST data. Since when
        # formatting the error the request handler might access
        # self.POST, set self._post and self._file to prevent
        # attempts to parse POST data again.
        # Mark that an error occurred. This allows self.__repr__ to
        # be explicit about it instead of simply representing an
        # empty POST
        self._mark_post_parse_error()
        raise
elif self.content_type == 'application/x-www-form-urlencoded':
    self._post, self._files = QueryDict(self.body, encoding=self._encoding), MultiValueDict()
else:
    self._post, self._files = QueryDict(encoding=self._encoding), MultiValueDict()
```
It looks to me like the current structure is to avoid `self._mark_post_parse_error()` more than once. This formulation is clearer to me though:
```python
if self.content_type == 'multipart/form-data':
    if hasattr(self, '_body'):
        # Use already read data
        data = BytesIO(self._body)
    else:
        data = self
    try:
        self._post, self._files = self.parse_file_upload(self.META, data)
    except MultiPartParserError:
        # An error occured while parsing POST data. Since when formatting
        # the error the request handler might access self.POST, set
        # self._post and self._file to prevent attempts to parse POST data
        # again. Mark that an error occured. This allows self.__repr__ to
        # be explicit about it instead of simply representing an empty POST.
        self._mark_post_parse_error()
        raise
elif self.content_type == 'application/x-www-form-urlencoded':
    try:
        content_length = int(self.META.get('HTTP_CONTENT_LENGTH', self.META.get('CONTENT_LENGTH', 0)))
    except (ValueError, TypeError):
        content_length = 0

    if (settings.DATA_UPLOAD_MAX_MEMORY_SIZE is not None and
            content_length > settings.DATA_UPLOAD_MAX_MEMORY_SIZE):
        self._mark_post_parse_error()
        raise RequestDataTooBig('Request body exceeded settings.DATA_UPLOAD_MAX_MEMORY_SIZE.')

    self._post, self._files = QueryDict(self.body, encoding=self._encoding), MultiValueDict()
else:
    self._post, self._files = QueryDict(encoding=self._encoding), MultiValueDict()
```